src/abstractdataframe/reshape.jl

```diff
@@ -200,23 +205,28 @@ function unstack(df::AbstractDataFrame, colkey::Int, value::Int)
     # group on anything not a key or value:
     g = groupby(df, setdiff(_names(df), _names(df)[[colkey, value]]))
     groupidxs = [g.idx[g.starts[i]:g.ends[i]] for i in 1:length(g.starts)]
-    rowkey = PooledDataArray(zeros(Int, size(df, 1)), [1:length(groupidxs);])
+    rowkey = zeros(Int, size(df, 1))
     for i in 1:length(groupidxs)
         rowkey[groupidxs[i]] = i
     end
-    keycol = PooledDataArray(df[colkey])
+    keycol = NullableNominalArray(df[colkey])
     valuecol = df[value]
     df1 = df[g.idx[g.starts], g.cols]
-    keys = unique(keycol)
+    keys = unique(keycol) # FIXME: why not levels?
```

Comment:
Yes, unused levels aren't removed, in particular for efficiency. Also, `unique` returns the levels in their order in appearance I think (which we could stop doing for efficiency if we don't think it's useful). But I left the FIXME because I didn't want to check whether it would be an issue to keep unused levels, nor to find the best way to check this. Now that tests pass it's time to investigate...